webnn.json

```diff
@@ -474,55 +474,26 @@
       },
       {
         "name": "build",
-        "args": [
-          {"name": "named operands", "type": "named operands"},
-          {"name": "callback", "type": "build graph callback"},
-          {"name": "userdata", "type": "void", "annotation": "*"}
-        ]
-      },
-      {
-        "name": "buildSync",
         "returns": "graph",
         "args": [
           {"name": "named operands", "type": "named operands"}
         ]
       }
     ]
   },
-  "build graph callback": {
-    "category": "callback",
-    "args": [
-      {"name": "status", "type": "build graph status"},
-      {"name": "graph", "type": "graph"},
-      {"name": "message", "type": "char", "annotation": "const*", "length": "strlen"},
-      {"name": "userdata", "type": "void", "annotation": "*"}
-    ]
-  },
-  "build graph status": {
-    "category": "enum",
-    "values": [
-        {"value": 0, "name": "success"},
-        {"value": 1, "name": "error"},
-        {"value": 2, "name": "context lost"},
-        {"value": 3, "name": "unknown"}
-    ]
-  },
   "input": {
     "category": "structure",
     "members": [
-      {"name": "buffer", "type": "void", "annotation": "const*", "length": "size"},
-      {"name": "size", "type": "size_t"},
-      {"name": "dimensions", "type": "int32_t", "annotation": "const*", "length": "dimensions count", "optional": true},
+      {"name": "resource", "type": "resource"},
+      {"name": "dimensions", "type": "int32_t", "annotation": "const*", "length": "dimensions count"},
```

Comment:
Because we still allow to set `dimensions` to be `nullptr`, should we keep it as optional?

Reply:
It's not align with Spec defination as optional and there is no differentiation with MLResource?

```
dictionary MLInput {
  required MLResource resource;
  required sequence<long> dimensions;
};
```

Reply:
I got it. However your implementation allows `dimensions` to be nullptr, and the examples actually use an `MLInput` as `MLResource` in this way. Does it? Otherwise, you probably need to implement two versions of NamedInputs, one accepts MLInput, another one accepts MLResource.

My point is if the nullptr is allowed and commonly used, how about setting nullptr as default value.

Reply:
> I got it. However your implementation allows `dimensions` to be nullptr, and the examples actually use an `MLInput` as `MLResource` in this way. Does it? Otherwise, you probably need to implement two versions of NamedInputs, one accepts MLInput, another one accepts MLResource.
>
> My point is if the nullptr is allowed and commonly used, how about setting nullptr as default value.

C function can't override, it make sense to set nullptr by default.

Reply:
sounds good